File: WAsP/Backend/WAsP/Models/Contract.py
from sqlalchemy import Column, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy import (Integer, String, Text )

from .models import Base
from .Productivity import Employees, Machines

class JobContract(Base):
    __tablename__ = "JobContract"

    id = Column(Integer, primary_key=True, autoincrement="auto")
    ContractID = Column(Integer)
    ContractName = Column(String)
    Details = Column(Text)

    tasks = relationship("TaskAssociation", back_populates="contract")

class TaskAssociation(Base):
    __tablename__ = "TaskAssociation"

    Jobid = Column(Integer, ForeignKey('JobContract.id'), primary_key=True)
    Taskid = Column(Integer, ForeignKey('TaskAssignment.id'), primary_key=True)

    task = relationship("TaskAssignment", back_populates="contracts")
    contract = relationship("JobContract", back_populates="tasks")



class TaskAssignment(Base):
    __tablename__ = "TaskAssignment"

    id = Column(Integer, primary_key=True, autoincrement="auto")
    WPSNo = Column(Integer, ForeignKey('WPS.id'))
    WelderID = Column(Integer, ForeignKey('Employees.payroll_id'))
    MachineID = Column(Integer, ForeignKey('Machines.id'))

    contracts = relationship("TaskAssociation", back_populates="task")

File: WAsP/Frontend/WAsP/src/app/_models/contract.ts
export class Contract {
    id: string;
    ContractName: string;
    Details: string;
}

export class TaskView {
    TaskID: string;
    WPS_No: string;
    FullName: string;
    MachineName: string;
}

export class Task {
    TaskID: string;
    WPS_No: string;
    WelderID: string;
    MachineID: string;
}

export class ContractTaskView {
    ContractID: string;
    ContractName: string;
    Details: String;
    TaskID: string;
    WPS_No: string;
    FullName: string;
    Description: string; //Machine Name
}